app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -519,16 +539,27 @@ public boolean onOptionsItemSelected(MenuItem inputItem) {
             builder.setView(customLayout);
 
             CheckBox ch = (CheckBox) customLayout.findViewById(R.id.checkBox_reverse);
+            CheckBox showArchived = (CheckBox) customLayout.findViewById(R.id.checkBox_archived);
+
             ch.setChecked(mOrderDirection == DBHelper.LoyaltyCardOrderDirection.Descending);
+            showArchived.setChecked(mArchiveFilter == DBHelper.LoyaltyCardArchiveFilter.All);
 
             builder.setSingleChoiceItems(R.array.sort_types_array, currentIndex.get(), (dialog, which) -> currentIndex.set(which));
 
             builder.setPositiveButton(R.string.sort, (dialog, which) -> {
-                if (ch.isChecked()) {
-                    setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Descending);
-                } else {
-                    setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Ascending);
+
+                if (ch.isChecked() && showArchived.isChecked())
+                    setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Descending, DBHelper.LoyaltyCardArchiveFilter.All);
+                else if(!ch.isChecked() && showArchived.isChecked()) {
+                    setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Ascending, DBHelper.LoyaltyCardArchiveFilter.All);
+                }
+                else if(ch.isChecked() && !showArchived.isChecked()){
+                    setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Descending, DBHelper.LoyaltyCardArchiveFilter.Active);
                 }
+                else if(!ch.isChecked() && !showArchived.isChecked()){
+                    setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Ascending, DBHelper.LoyaltyCardArchiveFilter.Active);
+                }
```

Comment:
I think this whole if/else can be simplified as something like:

```
setSort(
  loyaltyCardOrders.get(currentIndex.get()),
  ch.isChecked() ? DBHelper.LoyaltyCardOrderDirection.Descending : DBHelpder.LoyaltyCardOrderDirection.Ascending,
  showArchived.isChecked() ? DBHelper.LoyaltyCardArchiveFilter.All : DBHelper.LoyaltyCardArchiveFilter.Active
)
```

Given there's now 2 options, `ch` (short for checkbox) should be renamed because it is no longer the only checkbox. Something like `sortReversed` sounds good.